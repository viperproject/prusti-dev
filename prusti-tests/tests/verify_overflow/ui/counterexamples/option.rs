// compile-flags: -Punsafe_core_proof=true -Pcounterexample=true

use prusti_contracts::*;

struct X{
    a: Option<i32>,
}

#[ensures(!result)]
fn test_1(x: X) -> bool {
    match x.a {
        Some(2) => true,
        _ => false,
    }
}


fn main(){}