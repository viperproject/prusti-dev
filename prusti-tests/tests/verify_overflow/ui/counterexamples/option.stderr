error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/option.rs:9:11
   |
9  | #[ensures(!result)]
   |           ^^^^^^^
   |
note: the error originates here
  --> $DIR/option.rs:10:1
   |
10 | / fn test_1(x: X) -> bool {
11 | |     match x.a {
12 | |         Some(2) => true,
13 | |         _ => false,
14 | |     }
15 | | }
   | |_^
note: counterexample for "x"
       value:   X {
            a: std::option::Option::Some(
                2,
            ),
        }
  --> $DIR/option.rs:10:11
   |
10 | fn test_1(x: X) -> bool {
   |           ^
note: counterexample for "result"
       value:   true
  --> $DIR/option.rs:12:20
   |
12 |         Some(2) => true,
   |                    ^^^^

error: aborting due to previous error

